### PR TITLE
Use redis url if it exists

### PR DESCRIPTION
**Pull request by dwhenry**
This allows us to configure out Redis instance on Heroku which requires an username and password and automatically provide a `REDIS_URL` to do this.

### CODE REVIEW

**Comment by cbaines** [comment]
Looks good, I'm excited about being able to use REDIS_URL :) I'm happy for this to be merged, but I'd like someone else from GOV.UK to review it as well.

**Comment by dwhenry** [comment]
So I can't see why the travis build has failed as it just times out when I try and view the details.

**Comment by cbaines** [comment]
> So I can't see why the travis build has failed as it just times out when I try and view the details.

No problem, I triggered the GOV.UK CI system to build this branch. The reason why its failing seems to be this `app.root.basename.to_s`, on the CI system, this returns `"govuk_sidekiq_branches"`.

I'm not quite sure what to do in this case, one option would be to set GOVUK_APP_NAME in the tests, another option would be to make the tests independent of this, so that they pass with any value, as the functionality that you are adding is separate from this. Any thoughts?

If you do change the code, I'll need to trigger the CI build again.

```
Failures:

  1) GovukSidekiq::Railtie loads the SidekiqInitializer with the default config
     Failure/Error:
       SidekiqInitializer.setup_sidekiq(
         ENV["GOVUK_APP_NAME"] || app.root.basename.to_s,
         config
       )

       GovukSidekiq::SidekiqInitializer received :setup_sidekiq with unexpected arguments
         expected: ("govuk_sidekiq", {:host=>"127.0.0.1", :port=>6379})
              got: ("govuk_sidekiq_branches", {:host=>"127.0.0.1", :port=>6379})
       Diff:
       @@ -1,2 +1,2 @@
       -["govuk_sidekiq", {:host=>"127.0.0.1", :port=>6379}]
       +["govuk_sidekiq_branches", {:host=>"127.0.0.1", :port=>6379}]
     # ./lib/govuk_sidekiq/railtie.rb:15:in `block in <class:Railtie>'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `instance_exec'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `run'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:55:in `block in run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:54:in `run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/application.rb:352:in `initialize!'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `public_send'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `method_missing'
     # ./spec/railtie_spec.rb:18:in `block (2 levels) in <top (required)>'

  2) GovukSidekiq::Railtie when ENV contains REDIS_URL loads the SidekiqInitializer with the redis url
     Failure/Error:
       SidekiqInitializer.setup_sidekiq(
         ENV["GOVUK_APP_NAME"] || app.root.basename.to_s,
         config
       )

       GovukSidekiq::SidekiqInitializer received :setup_sidekiq with unexpected arguments
         expected: ("govuk_sidekiq", {:url=>#<Double (anonymous)>})
              got: ("govuk_sidekiq_branches", {:url=>#<Double (anonymous)>})
       Diff:
       @@ -1,2 +1,2 @@
       -["govuk_sidekiq", {:url=>#<Double (anonymous)>}]
       +["govuk_sidekiq_branches", {:url=>#<Double (anonymous)>}]
     # ./lib/govuk_sidekiq/railtie.rb:15:in `block in <class:Railtie>'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `instance_exec'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `run'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:55:in `block in run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:54:in `run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/application.rb:352:in `initialize!'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `public_send'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `method_missing'
     # ./spec/railtie_spec.rb:28:in `block (3 levels) in <top (required)>'

  3) GovukSidekiq::Railtie when ENV contains REDIS_HOST and REDIS_PORT loads the SidekiqInitializer with supplied details
     Failure/Error:
       SidekiqInitializer.setup_sidekiq(
         ENV["GOVUK_APP_NAME"] || app.root.basename.to_s,
         config
       )

       GovukSidekiq::SidekiqInitializer received :setup_sidekiq with unexpected arguments
         expected: ("govuk_sidekiq", {:host=>#<Double (anonymous)>, :port=>#<Double (anonymous)>})
              got: ("govuk_sidekiq_branches", {:host=>#<Double (anonymous)>, :port=>#<Double (anonymous)>})
       Diff:
       @@ -1,2 +1,3 @@
       -["govuk_sidekiq", {:host=>#<Double (anonymous)>, :port=>#<Double (anonymous)>}]
       +["govuk_sidekiq_branches",
       + {:host=>#<Double (anonymous)>, :port=>#<Double (anonymous)>}]
     # ./lib/govuk_sidekiq/railtie.rb:15:in `block in <class:Railtie>'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `instance_exec'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:30:in `run'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:55:in `block in run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/initializable.rb:54:in `run_initializers'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/application.rb:352:in `initialize!'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `public_send'
     # /home/jenkins/bundles/govuk_sidekiq_branches/ruby/2.3.0/gems/railties-5.0.1/lib/rails/railtie.rb:193:in `method_missing'
     # ./spec/railtie_spec.rb:40:in `block (3 levels) in <top (required)>'

Finished in 0.31841 seconds (files took 0.87202 seconds to load)
8 examples, 3 failures

Failed examples:

rspec ./spec/railtie_spec.rb:15 # GovukSidekiq::Railtie loads the SidekiqInitializer with the default config
rspec ./spec/railtie_spec.rb:25 # GovukSidekiq::Railtie when ENV contains REDIS_URL loads the SidekiqInitializer with the redis url
rspec ./spec/railtie_spec.rb:37 # GovukSidekiq::Railtie when ENV contains REDIS_HOST and REDIS_PORT loads the SidekiqInitializer with supplied details
```

**Comment by dwhenry** [comment]
@cbaines have modified the test, I think is a better approach with a named variable.

**Comment by cbaines** [comment]
> @cbaines have modified the test, I think is a better approach with a named variable.

I've triggered CI to process the updated branch, and as you should be able to see from GitHub, that has done the trick and the build now succeeds, thanks :)

File: lib/govuk_sidekiq/version.rb [hunk 1]
```diff
@@ -1,3 +1,3 @@
 module GovukSidekiq
-  VERSION = "1.0.1"
+  VERSION = "1.0.2"
```
Comment (tijmenb) [inline lib/govuk_sidekiq/version.rb]: We do version bumps in separate PRs to avoid conflicts with other branches (https://github.com/alphagov/styleguides/blob/master/rubygems.md#versioning).

File: README.md [hunk 2]
```diff
@@ -50,7 +50,9 @@ worker: bundle exec sidekiq -C ./config/sidekiq.yml
 ### 4. Configure puppet
 
 - Set `REDIS_HOST` and `REDIS_PORT` variables. `GOVUK_APP_NAME` should also be
-set, but this is already done by the default `govuk::app::config`.
+set, but this is already done by the default `govuk::app::config`. If you redis instance
```
Comment (tijmenb) [inline README.md]: Few typos: "if you**r** Redis instance requires more advance**d** connection settings".

File: README.md [hunk 3]
```diff
@@ -50,7 +50,9 @@ worker: bundle exec sidekiq -C ./config/sidekiq.yml
 ### 4. Configure puppet
 
 - Set `REDIS_HOST` and `REDIS_PORT` variables. `GOVUK_APP_NAME` should also be
-set, but this is already done by the default `govuk::app::config`.
+set, but this is already done by the default `govuk::app::config`. Ff your Redis instance
```
Comment (tijmenb) [inline README.md]: Sorry, a new typo: `Ff your Redis..`

**Comment by dwhenry** [comment]
@tijmenb @cbaines IS anything else required from me to get this change merged?

**Comment by tijmenb** [comment]
@dwhenry the "Ff your Redis" typo. The branch has merge conflicts now, so it should be rebased. After that it's good to merge!

**Comment by dwhenry** [comment]
@tijmenb Sorry about that second typo. 

This should all be ready to go now.

**Comment by tijmenb** [comment]
Cool, thanks. I'll let @cbaines merge (per our guidelines on external contributors).
